plugins/org.python.pydev.core/pysrc/_pydev_bundle/_pydev_completer.py

```diff
@@ -192,6 +192,17 @@ def completions_to_xml(completions):
     msg = ["<xml>"]
 
     for comp in completions:
+        temp_comp = comp
+        comp = []
+        
+        if IS_PY2:
```

Comment:
I think the patch could be replaced by a 2-liner (there's no need to create a new `comp` on python 3 -- also, I usually check the `__class__` in this case as `isinstance` has a considerable overhead and `unicode` subclasses aren't really expected):

```
if IS_PY2:
    comp = [(x.encode('utf-8') if x.__class__ == unicode else x) for x in comp]
```

Reply:
I agree with your suggestion, and will recommit it.Thanks and hope you will accept this pull request.